backend/inspirehep/records/api/literature.py

```diff
@@ -171,6 +172,36 @@ def update(self, data, disable_orcid_push=False, *args, **kwargs):
                 push_to_orcid(self)
             self.push_authors_phonetic_blocks_to_redis()
 
+    def get_modified_authors(self):
+        authors = get_value(self, "authors", [])
+        author_uuids_latest = set(get_value(self, "authors.uuid", []))
+        author_uuids_oldest = set(get_value(self._previous_version, "authors.uuid", []))
+        uuids_changed = set.symmetric_difference(
+            author_uuids_latest, author_uuids_oldest
+        )
+        changed_authors = []
+        for author in authors:
+            if author.get("uuid") in uuids_changed:
+                changed_authors.append(author)
+        author_ids_latest = set(
+            chain.from_iterable(get_value(self, "authors.ids.value", []))
+        )
+        author_ids_oldest = set(
+            chain.from_iterable(
+                get_value(self._previous_version, "authors.ids.value", [])
+            )
+        )
+        ids_changed = set.symmetric_difference(author_ids_latest, author_ids_oldest)
```

Comment:
the general idea is the same as for references, but I think you're copy/pasting too much here. From a high-level perspective, you want to find:
* new UUIDs (you don't really care about deletions as they're gone, so there's nothing to do for them)
* UUIDs for which the metadata has been changed (as an optimization you could check the relevant metadata only, but that's error prone, as can be seen here where you forgot emails, so let's not do that right now).

So it seems like you need some way to efficiently obtain an author metadata by UUID (at least for one of the copies, for the other one you can simply iterate on `authors`), that is a have `dict` mapping `uuid` -> author `dict`, let's call this `previous_authors_by_uuid`. You could then iterate on the current authors and for each of them:
* check if it exists in `previous_authors_by_uuid` or `previous_authors_by_uuid[uuid]` is different from the current one
* if so, `yield` it for disambiguation (it's better to use generators for this kind of stuff)